src/oracle/handler.js

```diff
@@ -80,6 +82,15 @@ exports.getParticipantsByTypeId = function (request, h) {
     idMap = participantCache.get(request.params.Type)
     if (idMap.get(request.params.ID)) {
       response = idMap.get(request.params.ID)
+      const currency = request.query.currency || undefined
+      const partySubIdOrType = request.query.partySubIdOrType || undefined
+      if (currency && partySubIdOrType) {
+        response = response.partyList.filter(party => party.partySubIdOrType === partySubIdOrType && party.currency === currency)
+      } else if (currency) {
+        response = response.partyList.filter(party => party.currency === currency)
+      } else if (partySubIdOrType) {
+        response = response.partyList.filter(party => party.partySubIdOrType === partySubIdOrType)
+      }
```

Comment:
please do a validation check of  
`request.query` before to check if it isnt undefined

Reply:
What was I thinking??